### PR TITLE
Linux build docs

### PR DESCRIPTION
**Pull request by dwoznicki**
This PR adds some more explicit instructions for building the maplibre-gl-native project on Debian (and to some extent, Ubuntu).

### CODE REVIEW

**Comment by wipfli** [comment]
Thanks @dwoznicki for taking the time to write this documentation. I followed your instructions on an ubuntu laptop but run into the following error
```
➜  maplibre-gl-native git:(main) ✗ cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10

-- Configuring GL-Native with OpenGL renderer backend
-- The following ICU libraries were not found:
--   i18n (optional)
-- Failed to find all ICU components (missing: ICU_INCLUDE_DIR ICU_LIBRARY) 
-- The following ICU libraries were not found:
--   uc (optional)
-- Failed to find all ICU components (missing: ICU_INCLUDE_DIR ICU_LIBRARY) 
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find JPEG (missing: JPEG_LIBRARY JPEG_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.16/Modules/FindJPEG.cmake:100 (find_package_handle_standard_args)
  platform/linux/linux.cmake:4 (find_package)
  CMakeLists.txt:1059 (include)
```

Any idea why this happens?

**Comment by wipfli** [comment]
Thanks @dwoznicki for taking the time to write this documentation. I followed your instructions on an ubuntu laptop but run into the following error
```
➜  maplibre-gl-native git:(main) ✗ cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10

-- Configuring GL-Native with OpenGL renderer backend
-- The following ICU libraries were not found:
--   i18n (optional)
-- Failed to find all ICU components (missing: ICU_INCLUDE_DIR ICU_LIBRARY) 
-- The following ICU libraries were not found:
--   uc (optional)
-- Failed to find all ICU components (missing: ICU_INCLUDE_DIR ICU_LIBRARY) 
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find JPEG (missing: JPEG_LIBRARY JPEG_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.16/Modules/FindJPEG.cmake:100 (find_package_handle_standard_args)
  platform/linux/linux.cmake:4 (find_package)
  CMakeLists.txt:1059 (include)
```

Any idea why this happens?

**Comment by acalcutt** [comment]
@wipfli  could that be missing submodules? I feel like I got a similar error when I did a regular git clone without --recurse-submodules 

In my post at https://github.com/maplibre/maplibre-gl-native/issues/280#issuecomment-1123907951 I mentioned using
`git clone --recurse-submodules -j8 https://github.com/maplibre/maplibre-gl-native.git`
to clone the repo, because If I just did a regulaar git clone it was missing all the submodules

**Comment by dwoznicki** [comment]
Yeah, if you haven't cloned all the submodules as @acalcutt mentioned, please go ahead and try that.

Also, it looks like I have `libjpeg8-dev` installed on my Ubuntu 20.04 machine, in addition to `libjpeg-turbo8`.

```
13:28:16 ❯ sudo apt list --installed | grep libjpeg

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

libjpeg-turbo8-dev/focal-security,focal-updates,now 2.0.3-0ubuntu1.20.04.1 amd64 [installed,automatic]
libjpeg-turbo8/focal-security,focal-updates,now 2.0.3-0ubuntu1.20.04.1 amd64 [installed,automatic]
libjpeg8-dev/focal,now 8c-2ubuntu8 amd64 [installed]
libjpeg8/focal,now 8c-2ubuntu8 amd64 [installed]
```

You could try `apt install libjpeg8-dev` as well. If that doesn't work, I'll see if I can get this working on a fresh install at some point.

**Comment by ntadej** [comment]
Should we also setup a CI pipeline?

**Comment by wipfli** [comment]
That sounds like a great idea, Tadej. I installed the missing packages with the wget links and then with standard apt on ubuntu and then it worked (still compiling). Thanks for the help!

**Comment by wipfli** [comment]
It worked :) This is my first native map of Switzerland:

out

Generated with 
```bash
./build/bin/mbgl-render --style style.json --output out.png --zoom 6 --lat 47.36649 --lon 8.53908
```

Thanks @dwoznicki and @acalcutt for contributing these Linux native guidelines!

**Comment by wipfli** [comment]
I would like to check if we can use the demo tiles and the style from there for this example. @acalcutt do you know how to package the files from https://github.com/maplibre/demotiles into an mbtiles file?

**Comment by acalcutt** [comment]
python3 mb-util --image_format=pbf /opt/demotiles/tiles demotiles.mbtiles

Remove the .txt

[demotiles.mbtiles.txt](https://github.com/maplibre/maplibre-gl-native/files/8837813/demotiles.mbtiles.txt)

**Comment by acalcutt** [comment]
@wipfli , while you are testing. I'd be interested to know if this label issue shows when rendering that way https://github.com/maplibre/maplibre-gl-native/issues/284

You could download the test data for that here https://github.com/acalcutt/tileserver-gl/releases/download/test_data/test_data_massachusetts.zip and the style wdb-osm is included in that.

**Comment by wipfli** [comment]
@dwoznicki do you think it is important to emphasis how to use a local mbtiles file? Because we could actually also just use the MapLibre demotiles style which uses the online demo tiles. Example:

```
wget https://raw.githubusercontent.com/maplibre/demotiles/gh-pages/style.json
./build/bin/mbgl-render --style style.json --output out.png
```
creates this nice map:

out

**Comment by acalcutt** [comment]
I think the local mbtiles was a good example. Good information I did not know. 

I like the from url example too.

**Comment by wipfli** [comment]
I just saw that the linux platform has a readme, see https://github.com/maplibre/maplibre-gl-native/tree/main/platform/linux Should we maybe update the linux platform readme instead of putting the info in the main readme?

**Comment by ntadej** [comment]
> I just saw that the linux platform has a readme, see https://github.com/maplibre/maplibre-gl-native/tree/main/platform/linux Should we maybe update the linux platform readme instead of putting the info in the main readme?

I agree, we should have platform specific instructions per platform and just link/refer to them from the main README.

**Comment by wipfli** [comment]
I think it is OK to replace the content of the platform linux README with something which actually works. Also, I like the idea of specifying the style URL in the command line. Thanks for you help, @dwoznicki!

File: platform/linux/README.md [hunk 1]
```diff
@@ -1,76 +1,162 @@
...
+This guide focuses on Debian and Ubuntu distributions.
 
-This process gives you a Linux desktop app built on a Linux host system.
+The build process should give you a set of `.a` files that you can use to include Maplibre GL Native in other C++ projects, as well as a set of executables that you can run to render map tile images and test the project.
 
-### Prerequisites
+## Prerequisites
 
-Install GCC 4.9+ if you are running Ubuntu 14.04 or older. Alternatively, you can also use [Clang 3.5+](http://llvm.org/apt/).
+The following dependencies are required to build Maplibre GL Native on Debian 11.
```
Comment (wipfli) [inline platform/linux/README.md:11]:
```suggestion
The following dependencies are required to build MapLibre GL Native on Debian 11.
```

File: platform/linux/README.md [hunk 2]
```diff
@@ -1,76 +1,162 @@
...
+
+If all went well, there should now be a `maplibre-gl-native/build/bin/mbgl-render` binary that you can run to generate map tile images. To test that it is working properly, run the following command.
+
+```bash
+./build/bin/mbgl-render --style https://raw.githubusercontent.com/maplibre/demotiles/gh-pages/style.json --output out.png
+```
+
+> I get an error `Error: Failed to open X display.` when I run this command.
+
+If you're setting up Maplibre GL Native on a headless server (i.e. one without a display), you'll need to simulate an X server to do any rendering.
```
Comment (wipfli) [inline platform/linux/README.md:56]:
```suggestion
If you're setting up MapLibre GL Native on a headless server (i.e. one without a display), you'll need to simulate an X server to do any rendering.
```

File: platform/linux/README.md [hunk 3]
```diff
@@ -1,76 +1,162 @@
...
+
+```bash
+xvfb-run -a ./build/bin/mbgl-render --style https://raw.githubusercontent.com/maplibre/demotiles/gh-pages/style.json --output out.png
+```
+
+This should produce an `out.png` map tile image with the default Maplibre styling from [the Maplibre demo](https://maplibre.org/).
```
Comment (wipfli) [inline platform/linux/README.md:62]:
```suggestion
This should produce an `out.png` map tile image with the default MapLibre styling from [the MapLibre demo](https://maplibre.org/).
```

**Review by wipfli — APPROVED** [review]
Really cool, thanks a lot @dwoznicki :)

**Comment by acalcutt** [comment]
Nice README. I'm glad you left in the local mbtiles stuff. sometimes the small things are hard to piece together.

**Pull request merged**